amazon/ionbenchmark/report.py

```diff
@@ -0,0 +1,89 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+import os
+import statistics
+from math import ceil
+
+from amazon.ionbenchmark.benchmark_runner import BenchmarkResult
+from amazon.ionbenchmark.benchmark_spec import BenchmarkSpec
+
+
+def report_stats(benchmark_spec: BenchmarkSpec, benchmark_result: BenchmarkResult, report_fields: list[str] = None):
+    """
+    Generate a report for the outcome of a running a benchmark.
+
+    Available fields:
+     * `name` – the name of the benchmark
+     * `operation` – the operation performed by the benchmark ('load', 'loads', 'dump', 'dumps')
+     * `file_size` – the size of the input data used in this benchmark
+     * `input_file` – the file used for this benchmark
+     * `format` – the format used for this benchmark
+     * `memory_usage_peak` – the peak amount of memory allocated while running the benchmark function
+     * `time_<stat>` – time statistic for the benchmark; `<stat>` can be `mean`, `min`, `max`, `median`, or `p<n>` where
+                       `<n>` is any number from 0 to 100 inclusive.
+     * `rate_<stat>` – throughput statistic for the benchmark; `<stat>` can be `mean`, `min`, `max`, `median`, or `p<n>`
+                       where `<n>` is any number from 0 to 100 inclusive.
+
+    :param benchmark_spec: The spec for the benchmark that was run
+    :param benchmark_result: The output from the benchmark
+    :param report_fields: List of fields to include in the report.
+    :return:
+    """
+    if report_fields is None:
+        report_fields = ['file_size', 'time_min', 'time_mean', 'memory_usage_peak']
+
+    result = {'name': benchmark_spec.get_name()}
+
+    for field in report_fields:
+        match field:
```

Comment:
Unless we're only doing the benchmarking for python 3.10+ we sadly can't use `match` I don't think there's a `__future__` import for it either (as it's a new syntax) but maybe.

Reply:
What happens if there's a change that improves `ion-python` when running on Python 3.8, but causes a regression on for Python 3.10 (or vice versa)? I would hope that we would prioritize the performance on the newer versions of Python.

Do we need to benchmark using multiple Python versions—is it actually important?

Reply:
I think it would be good to have the data for all the versions of python we claim to support. What we do with that, and how we act on it and what expectations we set we can talk about. Prioritizing newer versions makes sense as that will have the most long-term impact, but if a majority of our install base is using, say 3.9 (idk if they are) then we need to make sure we are doing work to positively impact them too.

Reply:
At the very least you need to either make this code work for all versions of python or change how the build works.

Reply:
Okay. I understand the issue now.